### PR TITLE
feat: add trace_filter method

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**
this should mark trace_filter as done here https://github.com/alloy-rs/alloy/issues/926

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
ty.

this should be moved to the traceApi trait instead and use &

**Comment by DoTheBestToGetTheBest** [comment]
> ty.
> 
> this should be moved to the traceApi trait instead and use &

ty so much for your fast review, changes has been made right now

File: crates/provider/src/ext/trace.rs [hunk 1]
```diff
@@ -56,6 +57,12 @@ where
         trace_type: &[TraceType],
     ) -> TransportResult<TraceResults>;
 
+    /// Traces matching given filter.
+    async fn trace_filter(
+        &self,
+        tracer: TraceFilter,
```
Comment (mattsse) [inline crates/provider/src/ext/trace.rs:63]:
```suggestion
        tracer: &TraceFilter,
```

File: crates/provider/src/ext/trace.rs [hunk 2]
```diff
@@ -112,6 +119,13 @@ where
         self.client().request("trace_transaction", (hash,)).await
     }
 
+    async fn trace_filter(
+        &self,
+        tracer: TraceFilter,
```
Comment (mattsse) [inline crates/provider/src/ext/trace.rs:124]:
```suggestion
        tracer: &TraceFilter,
```

**Review by mattsse — APPROVED** [review]

**Pull request merged**